src/iceberg/catalog/rest/resource_paths.cc

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "iceberg/catalog/rest/resource_paths.h"
+
+#include <format>
+
+#include "iceberg/catalog/rest/config.h"
+#include "iceberg/catalog/rest/endpoint_util.h"
+#include "iceberg/result.h"
+
+namespace iceberg::rest {
+
+Result<std::unique_ptr<ResourcePaths>> ResourcePaths::Make(
+    const RestCatalogConfig& config) {
+  // Validate and extract URI
+  auto it = config.configs().find(std::string(RestCatalogConfig::kUri));
```

Comment:
Can we add `Result<std::string> RestCatalogConfig::Uri() const` instead? Similarly, we can add getters for warehouse and prefix. It is not a good practice to expose internal logic (RestCatalogConfig) to the caller (ResourcePaths).